fog/view/server/src/server.rs

```diff
@@ -397,21 +403,31 @@ where
 
             // Insert the records into the enclave.
             self.add_records_to_enclave(
-                fetched_records.ingest_invocation_id,
+                fetched_records.ingress_key,
                 fetched_records.block_index,
                 fetched_records.records,
             );
         }
 
-        // Figure out the higehst fully processed block count and put that in the shared state.
-        let ingestable_ranges = self.db_fetcher.known_ingestable_ranges();
-        let missing_block_ranges = self.db_fetcher.known_missing_block_ranges();
+        // Figure out the highest fully processed block count and put that in the shared state.
+        let ingress_keys = self.db_fetcher.get_highest_processed_block_context();
+        let (highest_processed_block_count, reason_we_stopped) = self
+            .enclave_block_tracker
+            .highest_fully_processed_block_count(&ingress_keys);
 
         let mut shared_state = self.shared_state.lock().expect("mutex poisoned");
-        let highest_processed_block_count = self
-            .enclave_block_tracker
-            .highest_fully_processed_block_count(&ingestable_ranges, &missing_block_ranges);
-        shared_state.highest_processed_block_count = highest_processed_block_count;
+        if shared_state.highest_processed_block_count != highest_processed_block_count {
+            shared_state.highest_processed_block_count = highest_processed_block_count;
+            self.last_unblocked_at = Instant::now();
+        } else if self.last_unblocked_at.elapsed() >= Duration::from_secs(60) {
+            if let Some(reason_we_stopped) = reason_we_stopped {
+                log::warn!(self.logger, "We seem to be stuck at highest_processed_block_count = {} for at least a minute... we are blocked on an ingress key making progress: {:?}", highest_processed_block_count, reason_we_stopped);
```

Comment:
I suspect this will be noisy since our network is not very active right now. We probably want some kind of Prometheus alerting if highest processed block count is too far from the block count reported by nodes. Looping in @jgreat @joekottke in case they want to track this.

Reply:
maybe we should make a grpc API to get this info? like, highest processed block count and reason we stopped?

then we can get it when we need it and don't spam the logs?

Reply:
we could also make the number of seconds a config parameter, so it can be adjusted or turned off

Reply:
A grpc api endpoint for debugging sounds good. If we're okay making it user-facing then that should be an easy change. I can't think of an obvious reason why not.